### DISCUSSION TITLE
Windows Native collector, does not run as a Windows service

### DISCUSSION THREAD

**Discussion by StefanSa** [netdata/netdata#17658] [Q&A]
Hi there,
Thanks for the Windows Native collector, i have been waiting for it for a long time.
It is already running quite stable and i am really happy with it.
I have created a Windows msi package, this also includes the creation of the `netdata service`.
I noticed that starting the service with time out fails.
It looks like the `netdata.exe` does not contain / missing a `service API` that can be queried about the status of the service.
https://learn.microsoft.com/en-us/windows/win32/services/service-functions
Bad you can use the old Windows tools `SRVANY` or `Service Manager` as a wrapper for netdata.exe.

Have you experienced the same problems with the Windows service?

regards
Stefan

**Comment by StefanSa** [comment]
@ktsaou 
Γεια σου Κόστα και χαιρετισμούς στην Ελλάδα.
Have you also seen the problem with the Windows Service?
Thank you again for Netdata and for your time.

**Reply by ktsaou** [reply]
Sure. We are working on it. Will be there soon.

**Reply by StefanSa** [reply]
Thanks Costa for the information and good luck with netdata.
Τις καλύτερες ευχές από την κρύα και βροχερή Γερμανία.

**Reply by StefanSa** [reply]
@ktsaou 
But one more thing.
I see that you are using the complete msys2 installation ond windows,is that necessary?
I only take the necessary dll which works without problems so far. 

Dependencies can be easily determined.
copy your .EXE to a directory like build64/ 
`ldd build64/*|grep -iv system32|grep -vi windows|grep -v :$  | cut -f2 -d\> | cut -f1 -d\( | tr \\ / |while read a; do ! [ -e "build64/`basename $a`" ] && cp -v "$a" build64/; done`
This automatically determines the necessary dlls and copies them to the corresponding directory.

**Reply by ktsaou** [reply]
Exactly!

@vkalintiris the thing we were discussing...

**Reply by StefanSa** [reply]
You are actually only searching for the dependent dll that are in the same folder as the *.exe, this works fine without a complete msys2 inst.
The only thing missing is that netdata native runs as a service, but here you have to integrate the missing Windows sc api so that the status of the netdata service can be queried by sc. I am very positively surprised, how stable this is already running Costa.

**Comment by tradexsrl** [comment]
> Hi there, Thanks for the Windows Native collector, i have been waiting for it for a long time. It is already running quite stable and i am really happy with it. I have created a Windows msi package, this also includes the creation of the `netdata service`. I noticed that starting the service with time out fails. It looks like the `netdata.exe` does not contain / missing a `service API` that can be queried about the status of the service. https://learn.microsoft.com/en-us/windows/win32/services/service-functions Bad you can use the old Windows tools `SRVANY` or `Service Manager` as a wrapper for netdata.exe.
> 
> Have you experienced the same problems with the Windows service?
> 
> regards Stefan

Hi , where i can download the windows native collector ? i'm not able to find the .msi
